File: python/1800.Maximum-Ascending-Subarray-Sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        max_score = nums[0]
        cur_score = nums[0]

        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                cur_score += nums[i]
            else:
                cur_score = nums[i]
            max_score = max(max_score, cur_score)

        return max_score

File: javascript/1800.Maximum-Ascending-Subarray-Sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAscendingSum = function (nums) {
  let max_sum = nums[0];
  let cur_sum = nums[0];

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > nums[i - 1]) {
      cur_sum += nums[i];
    } else {
      cur_sum = nums[i];
    }

    max_sum = Math.max(max_sum, cur_sum);
  }

  return max_sum;
};
